src/github.com/matrix-org/dendrite/roomserver/types/types.go

```diff
@@ -1,10 +1,82 @@
 // Package types provides the types that are used internally within the roomserver.
 package types
 
+import (
+	"github.com/matrix-org/gomatrixserverlib"
+)
+
 // A PartitionOffset is the offset into a partition of the input log.
 type PartitionOffset struct {
 	// The ID of the partition.
 	Partition int32
 	// The offset into the partition.
 	Offset int64
 }
+
+// A StateKey is a pair of a numeric event type and a numeric state key.
+// It is used to lookup state entries.
+type StateKey struct {
+	// The numeric ID for the event type.
+	EventTypeNID int64
+	// The numeric ID for the state key or 0 if the event is not a state event.
```

Comment:
Somewhat weird that you can have a `StateKey` and it isn't a state event? What's the deal with that?